crates/relayer/relayer-components/src/relay/impls/packet_clearers/ack.rs

```diff
@@ -0,0 +1,109 @@
+use alloc::format;
+use cgp_core::async_trait;
+use core::fmt::Display;
+
+use crate::chain::traits::queries::packet_acknowledgements::CanQueryPacketAcknowledgements;
+use crate::chain::traits::queries::packet_commitments::CanQueryPacketCommitments;
+use crate::chain::traits::queries::send_packets::CanQuerySendPackets;
+use crate::chain::traits::queries::unreceived_acks_sequences::CanQueryUnreceivedAcksSequences;
+use crate::chain::types::aliases::{ChannelIdOf, PortIdOf};
+use crate::logger::traits::log::CanLog;
+use crate::relay::traits::chains::{CanRaiseRelayChainErrors, HasRelayChains};
+use crate::relay::traits::packet_clearer::PacketClearer;
+use crate::relay::traits::packet_relayer::CanRelayPacket;
+use crate::runtime::traits::runtime::HasRuntime;
+use crate::runtime::traits::task::{CanRunConcurrentTasks, Task};
+
+pub struct ClearAckPackets;
+
+pub struct RelayPacketTask<Relay>
+where
+    Relay: HasRelayChains,
+{
+    pub relay: Relay,
+    pub packet: Relay::Packet,
+}
+
+#[async_trait]
+impl<Relay> Task for RelayPacketTask<Relay>
+where
+    Relay: CanRelayPacket + CanLog,
+    Relay::Packet: Display,
+{
+    async fn run(self) {
+        if let Err(e) = self.relay.relay_packet(&self.packet).await {
+            self.relay.log_error(&format!(
+                "failed to relay packet the packet {} during ack packet clearing: {e:#?}",
+                self.packet
+            ));
+        }
+    }
+}
+
+#[async_trait]
+impl<Relay> PacketClearer<Relay> for ClearAckPackets
+where
+    Relay: Clone + CanRelayPacket + HasRuntime + CanRaiseRelayChainErrors + CanLog,
+    Relay::DstChain: CanQueryUnreceivedAcksSequences<Relay::SrcChain>
+        + CanQueryPacketAcknowledgements<Relay::SrcChain>,
+    Relay::SrcChain:
+        CanQueryPacketCommitments<Relay::DstChain> + CanQuerySendPackets<Relay::DstChain>,
+    Relay::Runtime: CanRunConcurrentTasks,
+    Relay::Packet: Display,
```

Comment:
I think the side here is mistaken. When clearing ack packets, we are querying the unreceived acks from the source chain. Remember that in a relay context, it is the destination chain that sends the ack, and the source chain that receives the ack.

Reply:
Oh indeed I got confused with the packet acknowledgements.
So the packet acknowledgements are queried from the destination chain https://github.com/informalsystems/hermes/blob/master/crates/relayer/src/chain/counterparty.rs?plain=1#L589, and then those sequence numbers are used to query the unreceived acks on the source chain https://github.com/informalsystems/hermes/blob/master/crates/relayer/src/chain/counterparty.rs?plain=1#L597.